### ISSUE TITLE
The start script may execute incorrectly WiFi Connect on slower networks

### ISSUE THREAD

**Issue by joshughes** [resin-io/resin-wifi-connect#173]
I am looking to use this as a setup tool... the issue is after I reboot I have set up via the captive portal again. 

It seems like there should be a flag to pass that would allow us to use https://developer.gnome.org/NetworkManager/stable/gdbus-org.freedesktop.NetworkManager.Settings.html#gdbus-method-org-freedesktop-NetworkManager-Settings.AddConnection

To save the connection permanently... so its a one-time setup and you don't have to worry about the device restarting.

**Comment by majorz** [comment]
The connections created through WiFi Connect persist across device reboots. We are using `AddAndActivateConnection` method for this, which is similar to `AddConnection`.

Can you please provide more details, so that we figure out why it is not working for you.

**Comment by joshughes** [comment]
Ya... So I have a little project where I setup a wifi hotspot off of an existing wifi connection... 

Wifi-Setup is working and persisting to disk... I can see the connection at 

/etc/NetworkManager/system-connections

but wifi-setup is still coming on and it seems like Network Manager is not using the connection or maybe its a timing issue where I don't give Network manager enough time to connect to wifi.

**Comment by joshughes** [comment]
I am using a Raspberry pi 3 and it seems like `resin-wifi-01` is the only network tried on boot... even though the other network was saved.

Resin.io doesn't let me create an image without SOME wifi network to connect to... but it doesn't seem like the pi will connect to anything except the first one.

**Comment by majorz** [comment]
You do not have to specify wireless connection settings when you download the resinOS image. You may choose `Ethernet Only` and this will not include the wireless connection profile. (this is when you use the dashboard web interface, but the same is true when downloading an image through our CLI ).

But even with wireless pre-configured you should still be able to use any other wireless connection defined through WiFi Connect.

**Comment by majorz** [comment]
@joshughes, did you find how to download `Ethernet Only` resinOS image from our dashboard? Also in case you still experience some strange issues like a wireless connection not activating, there is a button for providing support access on the resin.io dashboard and I can enter the test device and take a look to see what is going on.

**Comment by joshughes** [comment]
@majorz So the issue was I was not waiting long enough for the initial wifi connection to come up. 

I was using the `wget` example from the python project this repo links to and didn't think about the fact that the APP may start and check for the internet before the wifi connection was fully up. 

I wrote a loop that waits for 30 seconds on app startup checking for internet and that has solved the issue. 

Sorry for the confusion and I appreciate your quick response!

**Comment by majorz** [comment]
Right, it makes sense that to happen on a slower network. You may also check the our new [start script](https://github.com/resin-io/resin-wifi-connect/blob/master/scripts/start.sh) for other available conditions for running WiFi Connect as well.

**Comment by majorz** [comment]
I will reopen the issue and change its name, since I would like to investigate it further whenever possible

**Comment by iball8** [comment]
@joshughes could you explain how you implemented the to wait for 30 seconds? I am having issues connecting to internet. WiFi Connect is keeps running after I enter password and select ssid. I am certain that I am inputing my password correctly as well.
